src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -66,90 +90,133 @@ public void setFiles(FileCollection files) {
         this.files = files;
     }
 
-    @Input
+    /**
+     * @return algorithm
+     * @deprecated Use getAlgorithmProperty() instead
+     */
+    @Deprecated
+    @Internal
     public Algorithm getAlgorithm() {
+        return algorithm.get();
+    }
+
+    @Input
+    public Property<Algorithm> getChecksumAlgorithm() {
         return algorithm;
     }
 
     public void setAlgorithm(Algorithm algorithm) {
-        this.algorithm = algorithm;
+        this.algorithm.set(algorithm);
     }
 
-    @OutputDirectory
+    /**
+     * @return output directory
+     * @deprecated Use getOutputDirProperty() instead
+     */
+    @Deprecated
+    @Internal
     public File getOutputDir() {
+        return outputDir.get().getAsFile();
+    }
+
+    @OutputDirectory
+    public DirectoryProperty getOutputDirectory() {
         return outputDir;
     }
 
-    public void setOutputDir(File outputDir) {
-        if (outputDir.exists() && !outputDir.isDirectory()) {
+    public void setOutputDir(File outputDirAsFile) {
+        if (outputDirAsFile.exists() && !outputDirAsFile.isDirectory()) {
             throw new IllegalArgumentException("Output directory must be a directory.");
         }
-        this.outputDir = outputDir;
+        this.outputDir.set(outputDirAsFile);
     }
 
+    //TODO Migrate to InputChanges
     @TaskAction
     public void generateChecksumFiles(IncrementalTaskInputs inputs) throws IOException {
-        if (!getOutputDir().exists()) {
-            if (!getOutputDir().mkdirs()) {
-                throw new IOException("Could not create directory:" + getOutputDir());
+        File outputDirAsFile = getOutputDir();
```

Comment:
:x: Let's not use the now deprecated getters in the task action but the proper properties instead.